### PR TITLE
Replace travis by github action

### PR DESCRIPTION
**Pull request by gkrenn**
- move build of docker image from travis to github actions
- speed up build by 
   - executing tasks asynchronously 
   - removing ARM64 build from non release branches
   - caching dependencies
- arm64 builds are running in QEMU => no ARM64 machine required

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -8,15 +8,15 @@ ARG GO_BUILD_ARGS
 COPY . /app
 WORKDIR /app
 
-RUN go get github.com/google/go-licenses && go-licenses save ./... --save_path third_party_licenses --force
-RUN go get -d ./...
+# move previously cached go modules to gopath
+RUN mkdir -p ${GOPATH}/pkg && mv ./mod ${GOPATH}/pkg
```
Comment (chrismuellner) [inline Dockerfile:12]: this fails when running `make deploy-local-easy` for me, with the message that `./mod` doesn't exist

File: .github/workflows/dockerimagebuild.yml [hunk 2]
```diff
@@ -0,0 +1,217 @@
+name: run tests & build docker images
```
[the remark this answers is not in the document]
Comment (gkrenn) [reply .github/workflows/dockerimagebuild.yml:1]: So you would call the workflow and the file ci / ci.yml ?

File: .github/workflows/dockerimagebuild.yml [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+
+on:
```
[the remark this answers is not in the document]
Comment (gkrenn) [reply .github/workflows/dockerimagebuild.yml:3]: As we only want our docker image to run uploaddockerimage if the tests succeed, this is not possible.

Dependencies are only possible within one workflow

File: .github/workflows/dockerimagebuild.yml [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+    # The branches below must be a subset of the branches above
+    branches: [ master ]
+
+jobs:
+  tests:
+    name: Run unit tests
+    runs-on: ubuntu-latest
+    steps:
+      -
+        name: Checkout
```
[the remark this answers is not in the document]
Comment (gkrenn) [reply .github/workflows/dockerimagebuild.yml:17]: In my opinion, it's easier to separate the steps if there is a newline, but you are right, I should format them accordingly

File: .github/workflows/dockerimagebuild.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (gkrenn) [reply .github/workflows/dockerimagebuild.yml]: Renamed it

File: .github/workflows/dockerimagebuild.yml [hunk 5]
```diff
@@ -0,0 +1,217 @@
...
+          restore-keys: |
+            ${{ runner.os }}-go-build-
+      -
+        name: Prepare
+        id: prep
+        run: |
+          # https://blog.oddbit.com/post/2020-09-25-building-multi-architecture-im/ 
+          DOCKER_IMAGE=registry.gitlab.com/gabriel.krenn/test-private-registry/actions
+          VERSION=latest
+          SHORTREF=${GITHUB_SHA::8}
```
[the remark this answers is not in the document]
Comment (gkrenn) [reply .github/workflows/dockerimagebuild.yml]: Remainder of workflow merge, will delete it

File: .github/workflows/ci.yml [hunk 6]
```diff
@@ -0,0 +1,190 @@
+name: CI
+
+on:
+  push:
```
Comment (chrismuellner) [inline .github/workflows/ci.yml:4]:
```suggestion
on: push
```

**Review by chrismuellner — APPROVED** [review]

**Pull request merged**